go/vt/vtgate/planbuilder/testdata/filter_cases.txt

```diff
@@ -2126,7 +2126,25 @@ Gen4 plan same as above
     "Vindex": "music_user_map"
   }
 }
-Gen4 plan same as above
+{
+  "QueryType": "SELECT",
+  "Original": "select id from music where id is null",
+  "Instructions": {
+    "OperatorType": "Route",
+    "Variant": "Scatter",
+    "Keyspace": {
+      "Name": "user",
+      "Sharded": true
+    },
+    "FieldQuery": "select id from music where 1 != 1",
+    "Query": "select id from music where id is null",
+    "Table": "music",
+    "Values": [
+      "NULL"
+    ],
+    "Vindex": "music_user_map"
```

Comment:
Why are we using the vindex at all, if it is a scatter?

Reply:
we could clean it up and not use these at all. at the moment, we are just changing the opcode to scatter so they are valued much less and only used when no other alternative exists. You're right though, this is confusing. I'll clean it up.

/cc @arthurschreiber